[cxx-interop] Allow disabling Cxx build when building stdlib

egorzhdan opened the pull request:
This makes it possible to pass `SWIFT_BUILD_STDLIB_CXX_MODULE=FALSE` to CMake to disable Cxx and CxxStdlib modules when building the Swift stdlib or the SDK overlay.

rdar://108206321

egorzhdan committed:
[cxx-interop] Allow disabling Cxx build when building stdlib

This makes it possible to pass `SWIFT_BUILD_STDLIB_CXX_MODULE=FALSE` to CMake to disable Cxx and CxxStdlib modules when building the Swift stdlib or the SDK overlay.

rdar://108206321

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@swift-ci please smoke test

compnerd commented on cmake/caches/Windows-aarch64.cmake:
```diff
@@ -74,7 +74,6 @@ set(LLDB_ENABLE_LIBXML2 NO CACHE BOOL "")
 set(SWIFT_INCLUDE_DOCS NO CACHE BOOL "")
 set(SWIFT_BUILD_ENABLE_PARSER_LIB YES CACHE BOOL "")
 set(SWIFT_BUILD_STDLIB_EXTRA_TOOLCHAIN_CONTENT NO CACHE BOOL "")
-set(SWIFT_BUILD_STDLIB_CXX_MODULE NO CACHE BOOL "")
```
Why is this being removed?  We don't build any of the runtime components in the toolchain.

egorzhdan replied:
Looks like the swiftCxx module is still not built when I specify `SWIFT_BUILD_STDLIB_CXX_MODULE=YES` in the runtime presets.
```
LINK : fatal error LNK1104: cannot open file 'swiftCxx.lib'
```

egorzhdan replied:
I'm not super familiar with the build pipeline on Windows. Since the Cxx module is shipped with the toolchain, would it be acceptable to build it alongside the toolchain as well?

compnerd replied:
No, that wouldn't.  That would basically force the stdlib to *always* be required, and we cannot do that because it takes too long.

compnerd commented on cmake/caches/Windows-x86_64.cmake:
```diff
@@ -73,7 +73,6 @@ set(LLDB_EMBED_PYTHON_HOME NO CACHE BOOL "")
 set(SWIFT_INCLUDE_DOCS NO CACHE BOOL "")
 set(SWIFT_BUILD_ENABLE_PARSER_LIB YES CACHE BOOL "")
 set(SWIFT_BUILD_STDLIB_EXTRA_TOOLCHAIN_CONTENT NO CACHE BOOL "")
-set(SWIFT_BUILD_STDLIB_CXX_MODULE NO CACHE BOOL "")
```
Similar

compnerd reviewed (commented):
Could you please add the `SWIFT_BUILD_STDLIB_CXX_MODULE` as `YES` in the runtime configurations?

egorzhdan commented:
@swift-ci please smoke test

egorzhdan committed:
[cxx-interop] Allow disabling Cxx build when building stdlib

This makes it possible to pass `SWIFT_BUILD_STDLIB_CXX_MODULE=FALSE` to CMake to disable Cxx and CxxStdlib modules when building the Swift stdlib or the SDK overlay.

rdar://108206321

egorzhdan commented:
@swift-ci please smoke test

edymtt commented:
Duplicate of #65398
